src/tools/miri/tests/pass/btreemap.rs

```diff
@@ -50,7 +49,7 @@ pub fn main() {
     test_all_refs(&mut 13, b.values_mut());
 
     // Test forgetting the extractor.
-    let mut d = b.extract_if(.., |_, i| *i < 30);
+    let mut d = b.extract_if::<_, std::ops::RangeFull, String>(.., |_, i| *i < 30);
```

Comment:
Ah, interesting. I think this might be too much friction if users have to annotate the type when they're using `..` since that's a fairly common case.

Ideally we'd somehow type-inference here that K=Q if nothing is specified but I don't know of any way to do that. So I think that's too big a change for a stabilization PR.